issue#106: Add Modal Window

Danielpachecou opened the pull request:
Hello Team! @arturasmckwcz @nadavosa 

Add reusable Modal component

- Create Modal wrapper component with overlay and content container
- Support for closing via outside click (optional), Escape key, or close function
- Implement focus trap to keep keyboard navigation within modal
- Prevent body scroll when modal is open
- Provide close function to children via useModal hook or render prop
- Fully responsive with styled-components using project CSS variables
- Accessible with proper focus management and keyboard support

Location: src/components/core/modal/

Happy weekend!

Danielpachecou committed:
issue#106 add modal window

arturasmckwcz commented:
closes #106

arturasmckwcz commented:
@Danielpachecou we use `yarn` in our CI so default lock file is `yarn.lock` and `package-lock.json` is absolutely redundant. please remove it from the PR

arturasmckwcz commented on src/components/core/modal/Modal.tsx:
```diff
@@ -0,0 +1,204 @@
+  box-shadow: var(--n4d-box-shadow);
+  width: 100%;
+  padding: 24px;
+  /* Asegurar que el contenido del modal puede recibir eventos */
```
NIT: Despite the availability of translation tools everywhere, we prefer having comments and other info in English.

arturasmckwcz commented on src/components/core/modal/Modal.tsx:
```diff
@@ -0,0 +1,204 @@
+    if (!modalRef.current) return [];
+
+    const focusableSelectors = [
+      'a[href]',
```
@Danielpachecou somehow your local setup didn't run `pre-commit` hook.

please do that manually. it formates and lints the staged code before committing it.

in this case we use double quotes.

arturasmckwcz reviewed (commented):
thank you @Danielpachecou 
the code shines!
I only ask you to do proper formatting and translate comments to English

Danielpachecou committed:
fix: remove package-lock.json, translate comments to English, use double quotes

Danielpachecou committed:
Merge branch 'develop' into feat/106-modal-window

Danielpachecou commented:
Fixed all issues:
- Removed package-lock.json (npm habit, will check project setup)
- Translated comments to English
- Changed to double quotes

yarn conflict: My branch had an older `@types/node` version (20.19.24) while develop had the newer one (24.10.0). I resolved it by accepting develop's version to keep dependencies consistent, hope that is fine 

All changes committed and pushed.

arturasmckwcz reviewed (approved):
thank you!
